packages/app/obojobo-repository/shared/components/notification.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import { useState, useEffect } from 'react'
+import './notification.scss'
+
+const Notification = () => {
+	const [notifications, setNotifications] = useState([])
+	const [hiddenNotifications, setHiddenNotifications] = useState([])
+
+	useEffect(() => {
+		if (document && document.cookie) {
+			const cookiePropsRaw = decodeURIComponent(document.cookie).split(';')
+
+			let parsedValue
+			cookiePropsRaw.forEach(c => {
+				const parts = c.trim().split('=')
+				if (parts[0] === 'notifications') {
+					parsedValue = JSON.parse(parts[1])
+				}
+			})
+
+			const parsedNotifications = parsedValue
+			setNotifications(parsedNotifications)
+		} else {
+			//there is nothing to render
+		}
+	}, [])
+
+	//when user clicks exit button, remove notification from state and add to hidden notifications
+	function onClickExitNotification(key) {
+		setHiddenNotifications(prevHiddenNotifications => [...prevHiddenNotifications, key])
+		setNotifications(prevNotifications => prevNotifications.filter((_, index) => index !== key))
+	}
+
+	const renderNotification = (key, text, title) => {
+		return (
+			<div className={`notification-banner`} key={key}>
+				<div className="notification-header">
+					<h1>{title}</h1>
+					<button onClick={() => onClickExitNotification(key)} className="notification-exit-button">
+						x
+					</button>
+				</div>
+				<p>{text}</p>
+			</div>
+		)
+	}
+
+	//rewrite to cookie to remove notifications that have been hidden
+	if (hiddenNotifications && hiddenNotifications.length >= 1) {
+		const jsonNotifications = JSON.stringify(notifications)
+		const cookieString = `${encodeURIComponent(jsonNotifications)};`
+		document.cookie = 'notifications=' + cookieString
```

Comment:
These three lines can be moved into the `onClickExitNotification` function and the `hiddenNotifications` state variable can be eliminated entirely.

Reply:
The issue I encountered here is that if they are in the onClickExitNotification function then the notifications exit and remove from the page, however, they are re-rendered on page refresh and so the exit does not persist after refresh.  This was because on refresh the cookie is retaken and the state variables redefined. Any change to the state variable notifications is undone in refresh. However, if I exited two notifications the first one would be removed and only the second one would persist. So I created a list of notifications to be hidden called hidden notifications and added the if statement to update the cookie after the exit button is pressed, but before refreshing the page so that refreshing wouldn't lose the last exited change. This was a work-around so I will look into it more to see if there is a better solution.